### PR TITLE
6 player actions

### PR DESCRIPTION
**Pull request by houghta4**
Shooting an enemy decreases bullets. Must reload when empty
An enemy shot reduces health
Health packs can be used to heal
HUD additions:
 - live ammo count
 - reload word
 - live health/ health bar
 - health packs
 - heal word

### CODE REVIEW

File: src/game/player/systems.rs [hunk 1]
```diff
@@ -2,32 +2,108 @@ use bevy::prelude::*;
...
+                player.ammo.0 = 0;
+            }
+        }
+    }
+}
+
+/// Whenever a `PlayerReloadEvent` happens, the player's ammo will be replenished
+///
+/// This is triggered by the `Player` when a `WordTarget::Reload` is typed
+pub fn player_reload(
```
Comment (nnickodem) [inline src/game/player/systems.rs:79]: the word for this changes everytime an enemy is shot as well, is that intended? same for health pack

File: src/game/player/systems.rs [hunk 2]
```diff
@@ -2,32 +2,108 @@ use bevy::prelude::*;
...
+) {
+    for _ in reload_event_reader.iter() {
+        if let Ok(mut player) = player_q.get_single_mut() {
+            player.ammo.0 = player.ammo.1;
+            println!("<< reloading >>");
+        }
+    }
+}
+
+pub fn player_heal(
```
Comment (nnickodem) [inline src/game/player/systems.rs:91]: the next word for this doesnt appear until an enemy is shot, even when the player still has damage. Also the healthbar itself doesnt seem to refill

File: src/game/player/systems.rs [hunk 3]
```diff
@@ -2,32 +2,108 @@ use bevy::prelude::*;
...
+        }
+    }
+}
+
+/// Whenever a `EnemyShotEvent` happens, the player will lose ammo
+///
+/// This is triggered by the `Player` when a `WordTarget::Enemy` is typed
+pub fn player_shot_enemy(
+    mut player_q: Query<&mut Player>,
+    mut enemy_shot_event_reader: EventReader<EnemyShotEvent>,
```
Comment (nnickodem) [inline src/game/player/systems.rs:63]: would this run the risk of only one of the eventreaders for this event getting the event? feels iffy reading an event twice. Also enemies can still be shot when the ammo is 0

File: src/game/player/systems.rs [hunk 4]
```diff
@@ -2,32 +2,108 @@ use bevy::prelude::*;
...
+) {
+    for _ in enemy_shot_event_reader.iter() {
+        if let Ok(mut player) = player_q.get_single_mut() {
+            if player.ammo.0 > 0 {
+                player.ammo.0 -= 5; // TODO: subtract by 1 when finished testing
```
Comment (nnickodem) [inline src/game/player/systems.rs:68]: the ammo used should probably be different by gun, for some added slight realism

File: src/game/player/systems.rs [hunk 5]
```diff
@@ -2,32 +2,108 @@ use bevy::prelude::*;
...
+pub fn player_heal(
+    mut player_q: Query<&mut Player>,
+    mut heal_event_reader: EventReader<PlayerHealEvent>,
+) {
+    for _ in heal_event_reader.iter() {
+        if let Ok(mut player) = player_q.get_single_mut() {
+            if player.health_packs > 0 {
+                player.health_packs -= 1;
+
+                println!("healing!");
```
Comment (nnickodem) [inline src/game/player/systems.rs:100]: can remove this now

**Review by nnickodem — CHANGES_REQUESTED** [review]

File: src/game/player/systems.rs (hunk 3 above)
Comment (houghta4) [reply src/game/player/systems.rs:62]: It seems like events can be captured multiple times as long as each reader catches it within 2 updates. https://bevy-cheatbook.github.io/programming/events.html

File: src/game/player/systems.rs (hunk 4 above)
Comment (houghta4) [reply src/game/player/systems.rs:76]: I think this should be dealt with once we introduce multiple guns

**Comment by houghta4** [comment]
Words will now only update after they are typed. Eg, heal word refreshes after a heal, reload only after reloading.

Health packs heal to full

Health bar will work properly after healing

**Review by nnickodem — APPROVED** [review]

**Pull request merged**